course_catalog/tasks.py

```diff
@@ -35,26 +36,46 @@ def get_edx_data(force_overwrite=False):
         settings.EDX_API_URL
         and settings.EDX_API_CLIENT_ID
         and settings.EDX_API_CLIENT_SECRET
+        and settings.OCW_LEARNING_COURSE_BUCKET_NAME
+        and settings.OCW_LEARNING_COURSE_ACCESS_KEY
+        and settings.OCW_LEARNING_COURSE_SECRET_ACCESS_KEY
     ):
-        log.warning("Required settings missing for get_edx_data")
+        log.warning("Required settings missing for sync_and_upload_edx_data")
         return
     url = settings.EDX_API_URL
     access_token = get_access_token()
+    error_occurred = False
+
+    edx_data = {"count": 0, "catalog_url": settings.EDX_API_URL, "results": []}
 
     while url:
         response = requests.get(url, headers={"Authorization": "JWT " + access_token})
         if response.status_code == 200:
             for course_data in response.json()["results"]:
+                edx_data["results"].append(course_data)
                 try:
                     parse_mitx_json_data(course_data, force_overwrite)
                 except:  # pylint: disable=bare-except
-                    log.exception("Error encountered parsing MITx json")
+                    log.exception(
+                        "Error encountered parsing MITx json for %s", course_data["key"]
+                    )
         else:
             log.error("Bad response status %s for %s", str(response.status_code), url)
+            error_occurred = True
             break
 
         url = response.json()["next"]
 
+    if error_occurred:
```

Comment:
Don't know if you're still working on this, but a few comments:
1. This is the reverse of what we want. We only want to upload if there wasn't an error
1. It would be clearer to call this something like `should_upload_to_s3` and initialize it to `True`. If `parse_mitx_json_data`, that's technically an error, so it's confusing as to why you wouldn't set `error_occurred` to `True` when it fails.
1. A couple `log.info` lines would be helpful in this task. One to indicate that we're fetching and syncing edx data, and another one to indicate that we're uploading to S3. You would only need to log the latter if there wasn't an error

Reply:
@gsidebo Thanks for setting in to wrap up this PR!
1) Thanks---I saw that immediately but got pulled away from pushing right away. I also took the opportunity to add a new check to the unit tests in the same commit.
2) Happy to rename it...renamed!The reason I didn't set the flag to not upload on a  `parse_mitx_json_data` error is because you mentioned you think we should upload in your scenario case #2 and Peter said he could go either way. If we expect there to be at least one parsing error frequently due to a bad course, we may not want to block the upload of data for all the other courses. I'll add the flag for now to this case though, and it can be removed later if need be.
3) Added!

Let me know if you have any other PR comments.